security/fma-isthmus-l2-pectra.md

```diff
@@ -0,0 +1,281 @@
+
+
+### FM8: Smart contracts relying on sender check no longer ensures sender does not have code
+
+- **Description:** As part of EIP-7702, `msg.sender == tx.origin` no longer implies that the caller has no code. If unaddressed, this could cause
+some contracts to incorrectly assume the caller has no code.
+- **Risk Assessment:** Medium impact, low likelihood
+- **Mitigations:**
+  1. `isAddress` functions that are used as part of our predeployed contracts check code size, not the condition `msg.sender == tx.origin`.
+      - [OpenZeppelin reference](https://github.com/OpenZeppelin/openzeppelin-contracts-upgradeable/blob/0a2cb9a445c365870ed7a8ab461b12acf3e27d63/contracts/utils/AddressUpgradeable.sol#L41)
+  2. The checks, `msg.sender == tx.origin` and `tx.origin == msg.sender`, do not appear in the contracts currently used.
+- **Detection:** Some contracts that use the `msg.sender == tx.origin` check would allow delegated accounts to execute certain actions they weren't able to before.
+- **Recovery Path(s)**: Fix and upgrade the contracts.
+
+### Generic items we need to take into account:
+
+<!-- See [generic hardfork failure modes](./fma-generic-hardfork.md) and [generic smart contract failure modes](./fma-generic-contracts.md).
+Incorporate any applicable failure modes with FMA-specific mitigations and detections directly into this document. -->
+
+- [x] Check this box to confirm that these items have been considered and updated if necessary.
+
+## Action Items
+
+Below is what needs to be done before launch to reduce the chances of the above failure modes occurring, and to ensure they can be detected and recovered from:
+
+- [ ] Resolve all comments on this document and incorporate them into the document itself (Assignee: document author)
+- [ ] (NON-BLOCKING)  Make a way to enforce the gas limit of a block can't be lower of 30M (cf this [discord thread](https://discord.com/channels/1244729134312198194/1334603521454247946/1351651773755949117) with @tynes & @sebastianst) 
+- [x] (NON-BLOCKING):  Add the description of the new block header with the new field `request_hash` in this FMA (@meyer9)
+- [ ] (BLOCKING): Add a test that Validate a block with BLS accelerated precompile (@meyer9 ref to this [comment](https://github.com/ethereum-optimism/design-docs/pull/201/files#r1985623601))
+- [ ] (BLOCKING): Add a E2E test that Validate that minting to EOA that contains code (with Type4) is not an issue (@meyer9 assignee for now but could be someone else).
+- [ ]  (NON-BLOCKING): Identify path from the sequencer during the deposit transaction on the L2 that can cause unexpected behavior when an EOA has some code.
+- [ ]  (NON-BLOCKING): Indicate in this FMA reference the kurtosis devnet that allow the perform testing on Isthmus (L2 Pectra).
+- [ ]  (NON-BLOCKING): Tests to make sure that `ressources_limit` and `gas_limit` for each Superchain are not failling by overflowing from the config on L1 (assignee: @geoknee)
```

Comment:
I have performed this check manually [code here](https://github.com/ethereum-optimism/optimism/blob/371f5e08a9ff708b91af5bc7088e0238e7bf6ebf/op-node/rollup/superchain_test.go#L53-L83) and confirmed that all chains currently in the registry currently have a `gasLimit` of at least 30M and a `maxResourceLimit` of at least 20M. 

It wasn't worth committing this as a CI check, but I think we can add the link above and consider this item done.

Reply:
Thanks!
How to make sure we are running this just before the isthmus upgrade to make sure no systemconfig has been updated by mistake after we just ran the script?